feat: Add Homebrew formula for Open WebUI

cdamken opened the pull request:
# Pull Request: Add Homebrew Formula for Open WebUI

## 📋 Summary

This PR adds a fully tested and working Homebrew formula for installing Open WebUI on macOS and Linux systems via Homebrew.

## 🎯 What's Added

### Files Added:
- **`homebrew/open-webui.rb`** - Main Homebrew formula for Open WebUI v0.6.14
- **`homebrew/test-homebrew-formula.sh`** - Comprehensive test script
- **`homebrew/README.md`** - Documentation and usage instructions

## ✅ Features

- **Automatic Dependency Management**: Handles Python 3.12, Node.js 22, and ~98 Python packages
- **Virtual Environment Isolation**: Creates isolated Python environment to avoid conflicts
- **Clean Installation/Uninstallation**: Proper cleanup when removing
- **Homebrew Best Practices**: Follows official Homebrew formula conventions
- **Comprehensive Testing**: Automated test suite validates all functionality

## 🧪 Testing Results

All tests pass successfully:

- ✅ **Formula Syntax**: Ruby syntax validation passes
- ✅ **Installation**: Successfully installs Open WebUI v0.6.14 (2.6GB, 80,500 files)
- ✅ **CLI Functionality**: All commands work (`open-webui --help`, `open-webui serve`, etc.)
- ✅ **Uninstallation**: Clean removal of all files and dependencies
- ✅ **Reinstallation**: Works perfectly after uninstallation
- ✅ **Homebrew Built-in Tests**: Passes `brew test open-webui`

## 📦 Installation Details

**Dependencies:**
- **Runtime**: Python 3.12
- **Build**: Node.js 22
- **Python packages**: ~98 packages installed automatically via pip

**Installation Size**: ~2.6GB (80,500 files)

## 🚀 Usage

```bash
# Install
cd path/to/open-webui/homebrew
brew install --formula ./open-webui.rb

# Verify
open-webui --help

# Run server
open-webui serve
```

## 🔗 Related Issues

Addresses multiple community requests for native Homebrew installation support.

## ✅ Testing Instructions

To test this PR:

```bash
# Navigate to the homebrew directory
cd homebrew

# Run comprehensive tests
./test-homebrew-formula.sh
```

The test script automatically validates:
1. Formula syntax
2. Installation process
3. CLI functionality
4. Uninstallation process  
5. Reinstallation process

## 📝 Notes

- This formula is ready for submission to the official Homebrew repository
- All Python dependencies are managed automatically via pip in an isolated virtualenv
- The formula follows Homebrew best practices and conventions
- Clean installation and uninstallation tested extensively

---

## 🏷️ Type of Change

- [x] New feature (Homebrew formula)
- [x] Documentation updates
- [x] Testing improvements

## 📋 Checklist

- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Documentation updated
- [x] Tests added and passing
- [x] No breaking changes
- [x] Ready for review

Commit:
feat: Add Homebrew formula for Open WebUI

- Add homebrew/open-webui.rb: Complete Homebrew formula with dependencies
- Add homebrew/README.md: Installation and usage documentation
- Add homebrew/test-homebrew-formula.sh: Comprehensive test script

Formula features:
- Installs Python 3.11+ and Node.js 18+ dependencies
- Sets up virtual environment and builds frontend
- Creates proper CLI executable
- Handles static assets correctly
- Provides clean uninstall process

All tests pass successfully.

Classic298 commented:
please stick to the PR form and follow it. You did not agree to the CLA as per the PR form and you are trying to merge into the wrong branch too

Classic298 commented:
and your additions to the documentations would probably fit better into open-webui/docs repository

cdamken commented:
Thank you @Classic298 for the feedback on this PR. You were absolutely right about the issues:

1. ❌ **Wrong target branch**: This PR was targeting `main` instead of `dev`
2. ❌ **Missing CLA agreement**: The PR didn't include the required CLA statement
3. ❌ **Wrong PR format**: Didn't follow the official PR template
4. ❌ **Documentation location**: Good point about potentially using the docs repository

## ✅ **Resolution: New Proper PR Created**

I've addressed all these issues and created a new, properly formatted PR:

**🔗 New PR: #14983** - https://github.com/open-webui/open-webui/pull/14983

### What's Fixed in the New PR:

✅ **Target Branch**: Now correctly targets `dev` branch  
✅ **CLA Agreement**: Includes explicit CLA agreement statement  
✅ **PR Template**: Follows the official `.github/pull_request_template.md` format  
✅ **Proper Branch Name**: Uses `feat/homebrew-formula-proper` instead of `feature/homebrew-formula`  
✅ **Complete Checklist**: All required checklist items checked and completed  
✅ **Changelog Entry**: Proper "Keep a Changelog" format included  
✅ **Dependencies Documentation**: All new dependencies listed and explained  
✅ **Testing Documentation**: Comprehensive testing instructions and results  

### Additional Improvements:

- 🧪 **Enhanced Testing**: All tests still pass (formula syntax, installation, CLI functionality, uninstallation, reinstallation)
- 📚 **Better Documentation**: More detailed explanation of features and usage
- 🔒 **Security Notes**: Virtual environment isolation explained
- 🏗️ **Build Process**: Dependency management documented

I'm closing this PR in favor of the new, properly formatted one. Thank you for helping ensure we follow the project's contribution guidelines properly! 

**Action Items Completed:**
- [x] New PR created with correct target branch (`dev`)
- [x] CLA agreement included
- [x] Official PR template followed
- [x] All testing completed and documented
- [x] Dependencies properly documented

The new PR #14983 should now meet all the requirements for review and approval.

cdamken commented:
Closing in favor of properly formatted PR #14983 that addresses all feedback.
